cdap-app-fabric/src/main/java/co/cask/cdap/metadata/FieldLineageAdmin.java

```diff
@@ -224,16 +225,45 @@ FieldLineageDetails getOperationDetails(Constants.FieldLineage.Direction directi
 
     List<FieldOperationInfo> fieldOperationInfos = new ArrayList<>();
     for (Operation operation : orderedOperations) {
-        fieldOperationInfos.add(convertToFieldOperationInfo(operation));
+      fieldOperationInfos.add(convertToFieldOperationInfo(operation));
     }
 
     return fieldOperationInfos;
   }
 
+  /**
+   * Sort the operations in topological order. In topological order, each operation in the list
+   * is guaranteed to occur before any other operation, which reads its outputs.
```

Comment:
nit: 'operation, which reads' -> 'operation that reads'

Reply:
fixed.